corehq/apps/app_manager/suite_xml/sections/menus.py

```diff
@@ -43,11 +43,12 @@ def get_module_contributions(self, module):
                 menu = Menu(**menu_kwargs)
 
             def get_commands():
-                for form in module.get_forms():
-                    command = Command(id=id_strings.form_command(form))
+                forms = module.source_module.get_forms() if module.module_type == 'shadow' else module.get_forms()
```

Comment:
It's only twice (did think about it, and might change it, I agree this is clunky).